src/features/health/region.controller.js

```diff
@@ -5,36 +5,40 @@ const createRegion = async (req, res) => {
     const regionName = req.query.region;
     console.log(`Checking for existing region: ${regionName}`);
 
-    // checking to see if region exist
+    // Checking to see if region already exists
     const existingRegion = await Region.findOne({region: regionName});
     console.log(`Existing region found: ${existingRegion}`);
 
     if (existingRegion) {
         return res.status(400).send({ message: "Region already exists"});
     }
 
-    // creating new region
+    // Creating new region
     const newRegion = new Region({name: regionName});
     await newRegion.save();
 
-    //get top agents
-    const getTopAgents = async () => {
-        try {
-            const topAgents = await getTopAgentsBySales();
-            console.log('Top agents by sales:', topAgents);
-        } catch (error) {
-            console.error('Failed to fetch top agents:', error);
-        }
-    };
-    
-    getTopAgents();
-
-    //return region object
+    // Get top agents and assign them to the new region
+    try {
+        const topAgents = await getTopAgentsBySales();
+        console.log('Top agents by sales:', topAgents);
+
+        // Assuming getTopAgentsBySales returns an array of agent IDs
+        newRegion.top_agents = topAgents;
```

Comment:
Cool, I see the progress here. I like the approach of handling each piece of the puzzle individually.